content/docs/07-reference/01-ai-sdk-core/80-smooth-stream.mdx

```diff
@@ -42,14 +42,60 @@ const result = streamText({
     },
     {
       name: 'chunking',
-      type: '"word" | "line" | RegExp',
+      type: '"word" | "line" | RegExp | (buffer: string) => string | undefined | null',
       isOptional: true,
       description:
-        'Controls how the text is chunked for streaming. Use "word" to stream word by word (default), "line" to stream line by line, or provide a custom RegExp pattern for custom chunking.',
+        'Controls how the text is chunked for streaming. Use "word" to stream word by word (default), "line" to stream line by line, or provide a custom callback or RegExp pattern for custom chunking.',
     },
   ]}
 />
 
+#### Word chunking caveats with non-latin languages
+
+<Note>
+    The word based chunking **does not work well** with the following languages that do not delimit words with spaces:
+
+    For these languages the `word` option splits on the entire unicode range:
+
+    - Chinese - `\u4E00-\u9FFF`
+    - Japanese - `\u3040-\u309F` & `\u30A0-\u30FF`
+
+    For these languages you could pass your own language aware chunking function:
+
+    - Vietnamese
+    - Thai
+    - Javanese (Aksara Jawa)
+
+</Note>
+
+#### Regex based chunking
+
+To use regex based chunking, pass a `RegExp` to the `chunking` option.
+
+```ts
+// To split on underscores:
+smoothStream({
+  chunking: /_+/,
+});
+
+// To split on newlines:
+smoothStream({
+  chunking: /\n+/,
+});
+```
+
+#### Custom callback chunking
+
+To use a custom callback for chunking, pass a function to the `chunking` option.
+
+```ts
+smoothStream({
+  chunking: buffer => {
+    return buffer.match(/[^\n]+/)?.[0];
+  },
```

Comment:
need a better example - here they can just use regexp

Reply:
i was trying to think how they'd practically use it, i thought showing a regex would be the best option. What other example would you suggest?

Reply:
aha findIndexOf may be better